### PR TITLE
Improvement/EPMRPP-34927 React. Permission map

### PR DESCRIPTION
**Pull request by AmsterGet**

### CODE REVIEW

File: app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx [hunk 1]
```diff
@@ -23,110 +17,102 @@ export class PermissionMap extends Component {
...
-    const roles = [ADMINISTRATOR, PROJECT_MANAGER, MEMBER, OPERATOR, CUSTOMER];
-    const hasPermission = (
-      <td className={cx('col', 'has-permission')}>
-        <i className={cx('rp-icons', 'rp-icons-check')} />
-      </td>
+    const { intl } = this.props;
+    const roles = [PROJECT_MANAGER, MEMBER, OPERATOR, CUSTOMER];
+    const ownerPermission = (
+      <div title={intl.formatMessage(ROLE_NAMES.ownTitle)} className={cx('permission-container')}>
+        <div className={cx('owner-permission')} />
```
Comment (yana95) [inline app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx]: You don't need to create one more empty DIV. Just use `::before` or` ::after`

File: app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx [hunk 2]
```diff
@@ -23,110 +17,102 @@ export class PermissionMap extends Component {
...
+      </div>
     );
-    const noPermission = <td className={cx('col')} />;
+    const notOwnerPermission = (
+      <div
+        title={intl.formatMessage(ROLE_NAMES.notOwnTitle)}
+        className={cx('permission-container')}
+      >
+        <div className={cx('not-owner-permission')} />
```
Comment (yana95) [inline app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx]: You don't need to create one more empty DIV. Just use ::before or::after

File: app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx [hunk 3]
```diff
@@ -23,110 +17,102 @@ export class PermissionMap extends Component {
...
+        }
       }
-      return <Fragment key={`${role}_${permission}`}>{noPermission}</Fragment>;
+      return (
+        <td className={cx('permission-cell')} key={`${role}_${permission}`}>
+          <div className={cx('cell-content')}>
+            {permissionsArray.map((item, index) => (
+              // eslint-disable-next-line react/no-array-index-key
+              <Fragment key={index}>{item}</Fragment>
+            ))}
```
Comment (yana95) [inline app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx]: You don't need create `permissionsArray` to store permissions to render them using `map`. You can just 
```
{ PERMISSIONS_MAP[role][permission] && ownerPermission }
{ PERMISSIONS_MAP[role][permission] === ALL && notOwnerPermission }
```

File: app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx [hunk 4]
```diff
@@ -1,16 +1,10 @@
...
-import {
-  PROJECT_MANAGER,
-  OPERATOR,
-  CUSTOMER,
-  MEMBER,
-  ADMINISTRATOR,
-} from 'common/constants/projectRoles';
-import { ACTIONS, PERMISSIONS_MAP, ALL } from 'common/constants/permissions';
+import { PROJECT_MANAGER, OPERATOR, CUSTOMER, MEMBER } from 'common/constants/projectRoles';
+import { PERMISSIONS_MAP, ALL } from './permissions';
```
Comment (yana95) [inline app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx]: Please import `ALL` constant from `common\constants\permissions.js`

File: app/src/components/main/modal/modalLayout/modalContent/modalContent.jsx [hunk 5]
```diff
@@ -5,14 +5,14 @@ import styles from './modalContent.scss';
 
 const cx = classNames.bind(styles);
 
-export const ModalContent = ({ children, fullWidthContent }) => (
-  <div className={cx('modal-content', { 'full-width': fullWidthContent })}>{children}</div>
+export const ModalContent = ({ children, stretchedContent }) => (
```
Comment (yana95) [inline app/src/components/main/modal/modalLayout/modalContent/modalContent.jsx]: Now we don't need  stretchedContent prop. Remove it please.

File: app/src/components/main/modal/modalLayout/modalContent/modalContent.scss [hunk 6]
```diff
@@ -4,7 +4,6 @@
   font-family: $FONT-REGULAR;
   font-size: 14px;
 }
-.full-width {
-  padding: 25px 0 0 0;
-  margin: 0 -20px 0 -20px;
+.stretched-content {
+  padding: 30px 0;
```
Comment (yana95) [inline app/src/components/main/modal/modalLayout/modalContent/modalContent.scss]: Remove this class

File: app/src/components/main/modal/modalLayout/modalLayout.jsx [hunk 7]
```diff
@@ -10,13 +10,16 @@ import styles from './modalLayout.scss';
...
+  null,
+  {
+    hideModalAction,
+  },
+)
 export class ModalLayout extends Component {
   static propTypes = {
     className: PropTypes.string,
-    fullWidthContent: PropTypes.bool,
+    stretchedContent: PropTypes.bool,
```
Comment (yana95) [inline app/src/components/main/modal/modalLayout/modalLayout.jsx]: Remove this prop

File: app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissions.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
+import { PROJECT_MANAGER, OPERATOR, CUSTOMER, MEMBER } from 'common/constants/projectRoles';
+
+export const ALL = 'ALL';
+export const OWNER = 'OWNER';
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissions.js]: Done

File: app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx (hunk 3 above)
Comment (AmsterGet) [reply app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx]: Fixed

File: app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx (hunk 4 above)
Comment (AmsterGet) [reply app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx]: Fixed

File: app/src/components/main/modal/modalLayout/modalContent/modalContent.jsx (hunk 5 above)
Comment (AmsterGet) [reply app/src/components/main/modal/modalLayout/modalContent/modalContent.jsx]: Fxied

File: app/src/components/main/modal/modalLayout/modalContent/modalContent.scss (hunk 6 above)
Comment (AmsterGet) [reply app/src/components/main/modal/modalLayout/modalContent/modalContent.scss]: Fxied

File: app/src/components/main/modal/modalLayout/modalLayout.jsx (hunk 7 above)
Comment (AmsterGet) [reply app/src/components/main/modal/modalLayout/modalLayout.jsx]: Fxied

File: app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx (hunk 1 above)
Comment (AmsterGet) [reply app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx]: Fixed

File: app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx (hunk 2 above)
Comment (AmsterGet) [reply app/src/pages/inside/membersPage/modals/permissionMapModal/permissionMap/permissionMap.jsx]: Fixed

**Pull request merged**